scripts/log-session.py

```
@@ -0,0 +1,334 @@
+
+
+def get_memory_dir() -> Path:
+    """Get memory directory path, creating if needed."""
+    cwd = Path.cwd()
+    while cwd != cwd.parent:
+        if (cwd / 'CLAUDE.md').exists():
+            memory_dir = cwd / 'memory'
+            memory_dir.mkdir(exist_ok=True)
+            return memory_dir
+        cwd = cwd.parent
+    raise FileNotFoundError("Could not find project root (CLAUDE.md)")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)